testing/trino-testing-containers/src/main/java/io/trino/testing/containers/ReportLeakedContainers.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.testing.containers;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.model.Container;
+import io.airlift.log.Logger;
+import org.junit.platform.launcher.TestExecutionListener;
+import org.junit.platform.launcher.TestPlan;
+import org.testcontainers.DockerClientFactory;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.lang.Boolean.getBoolean;
+import static java.util.stream.Collectors.joining;
+
+public class ReportLeakedContainers
+        implements TestExecutionListener
+{
+    private static final Logger log = Logger.get(ReportLeakedContainers.class);
+    private static final boolean DISABLED = getBoolean("ReportLeakedContainers.disabled");
+
+    @Override
+    public void testPlanExecutionFinished(TestPlan testPlan)
+    {
+        if (DISABLED) {
+            log.info("ReportLeakedContainers disabled");
+            return;
+        }
+        log.info("Checking for leaked containers");
+
+        try (DockerClient client = DockerClientFactory.instance().client()) {
+            List<Container> containers = client.listContainersCmd()
```

Comment:
are containers immediately removed - are we sure those are gone by now?

Reply:
yes, normally they are removed with equivalent of `docker rm -f`, so immediate